app/components/billing-process.js

```diff
@@ -0,0 +1,184 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { not, filterBy, mapBy, equal } from '@ember/object/computed';
+
+const STEPS = {
+  stepOne: 'stepOne',
+  stepTwo: 'stepTwo'
+};
+
+let availablePlans = [
+  {
+    name: 'Bootstrap',
+    enabled: true
+  },
+  {
+    name: 'Startup',
+    enabled: true,
+    isDefault: true
+  },
+  {
+    name: 'Small Business',
+    enabled: true
+  },
+  {
+    name: 'Premium',
+    enabled: true
+  }
+];
+
+let defaultFormValues = {
```

Comment:
Yeah, I would suggest to use a bit different approach here. The proper "Ember Way" for handling such cases is:
- Wrap data in Ember Data models
- Fetch/create models in Routes (`model` hook)
- Bind data models to controller (`setupController` hook)
- Operate with the data in controller or bypass to child components
- Modify data models in actions

In this case, [`subscription` is a model](https://github.com/travis-ci/travis-web/blob/ep-subscription-billing/app/models/subscription.js), [`billing-info` is a model](https://github.com/travis-ci/travis-web/blob/ep-subscription-billing/app/models/billing-info.js), [`credit-card-info` is a model](https://github.com/travis-ci/travis-web/blob/ep-subscription-billing/app/models/credit-card-info.js). So you could create them all in the `model` hook of the route:
```js
model() {
  const billingInfo = this.store.createRecord('billing-info');
  const creaditCardInfo = this.store.createRecord('credit-card-info');
  return this.store.createRecord('subscription', { billingInfo, creditCardInfo });
}
```

You can define default values in the corresponding `attr`s in model definition. 

You can bind the model properties directly to form elements and mutate/update them from there. 

When it's time to save changes to server, in the task you can just:

```js
save: task(function* () {
  return this.subscription.save();
})
```

If you'd need to revert the data to defaults you can either [rollback attributes](https://api.emberjs.com/ember-data/3.10/classes/DS.Model/methods/rollbackAttributes?anchor=rollbackAttributes) for all related records or completely [`unloadRecord`](https://api.emberjs.com/ember-data/3.10/classes/DS.Model/methods/unloadRecord?anchor=unloadRecord) and re-create it from scratch. 

Also, worth mentioning, that when you create new model instances in route, you should also take care of cleaning them up when transitioning out of this route, e.g. to unload all unsaved models from store, because they can affect other parts of the app (show app in the lists, etc.).

Here's a Guides article on using Ember Data models: https://guides.emberjs.com/v3.7.0/models/creating-updating-and-deleting-records/